OpenRA.Mods.Common/Traits/World/PaletteFromGimpOrJascFile.cs

```diff
@@ -0,0 +1,131 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2018 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.IO;
+using OpenRA.Graphics;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Load a GIMP .gpl or JASC .pal palette file. Supports per-color alpha.")]
+	class PaletteFromGimpOrJascFileInfo : ITraitInfo
+	{
+		[FieldLoader.Require, PaletteDefinition]
+		[Desc("Palette name used internally.")]
+		public readonly string Name = null;
+
+		[Desc("Defines for which tileset IDs this palette should be loaded.",
+			"If none specified, it applies to all tileset IDs not explicitly excluded.")]
+		public readonly HashSet<string> Tilesets = new HashSet<string>();
+
+		[Desc("Don't load palette for these tileset IDs.")]
+		public readonly HashSet<string> ExcludeTilesets = new HashSet<string>();
+
+		[FieldLoader.Require]
+		[Desc("Name of the file to load.")]
+		public readonly string Filename = null;
+
+		[Desc("Map listed indices to shadow. Ignores previous color.")]
+		public readonly int[] ShadowIndex = { };
```

Comment:
Modders can define shadows (including alpha) directly in the palette, so this is redundant.